corporate/lib/stripe.py

```diff
@@ -2394,6 +2394,23 @@ def invoice_plan(self, plan: CustomerPlan, event_time: datetime) -> None:
             licenses_base = ledger_entry.licenses
 
         if invoice_item_created:
+            flat_discount, flat_discounted_months = self.get_flat_discount_info(plan.customer)
+            if flat_discounted_months > 0:
+                num_months = (
+                    12 if plan.billing_schedule == CustomerPlan.BILLING_SCHEDULE_ANNUAL else 1
+                )
+                flat_discounted_months = min(flat_discounted_months, num_months)
+                discount = flat_discount * flat_discounted_months
+                plan.customer.flat_discounted_months -= flat_discounted_months
+                plan.customer.save(update_fields=["flat_discounted_months"])
+                stripe.InvoiceItem.create(
+                    currency="usd",
+                    customer=plan.customer.stripe_customer_id,
+                    description=f"${flat_discount}/month new customer discount",
+                    # Negative value to apply discount.
+                    amount=(-1 * discount),
+                )
```

Comment:
Do we have `@transaction.atomic` around `invoice_plan`? reading this code, it would definitely be very nice if we could be certain that the parts of this happen transactionally, and I imagine that will be true for other parts of invoicing as well.

Reply:
Second commit.